File: src/main/java/io/xol/chunkstories/api/events/entity/EntityDeathEvent.java
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package io.xol.chunkstories.api.events.entity;

import io.xol.chunkstories.api.entity.Entity;
import io.xol.chunkstories.api.events.Event;
import io.xol.chunkstories.api.events.EventListeners;

/** This event is called upon confirmed death of a living entity. You can't and
 * shouldn't prevent it from dying here, instead use the EntityDamageEvent to
 * cancel the damage. */
public class EntityDeathEvent extends Event {
	// Every event class has to have this

	static EventListeners listeners = new EventListeners(EntityDeathEvent.class);

	@Override
	public EventListeners getListeners() {
		return listeners;
	}

	public static EventListeners getListenersStatic() {
		return listeners;
	}

	// Specific event code

	Entity entity;

	public EntityDeathEvent(Entity entity) {
		this.entity = entity;
	}

	public Entity getEntity() {
		return entity;
	}
}

File: src/main/java/xyz/chunkstories/api/events/entity/EntityDeathEvent.kt
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.events.entity

import xyz.chunkstories.api.entity.Entity
import xyz.chunkstories.api.events.Event
import xyz.chunkstories.api.events.EventListeners

/** This event is called upon confirmed death of a living entity. You can't and
 * shouldn't prevent it from dying here, instead use the EntityDamageEvent to
 * cancel the damage.  */
class EntityDeathEvent(entity: Entity) : Event() {

    override val listeners: EventListeners
        get() = listenersStatic

    // Specific event code

    var entity: Entity
        internal set

    init {
        this.entity = entity
    }

    companion object {
        // Every event class has to have this

        var listenersStatic = EventListeners(EntityDeathEvent::class.java)
            internal set
    }
}
